crates/dbsp/src/storage/file/cache.rs

```diff
@@ -51,106 +52,121 @@ impl CacheEntry for FileCacheEntry {
             Self::Data(data_block) => data_block.cost(),
         }
     }
-    fn from_read(raw: Arc<FBuf>, offset: u64, size: usize) -> Result<Self, Error> {
+}
+impl FileCacheEntry {
+    pub(super) fn from_read(
+        raw: Arc<FBuf>,
+        location: BlockLocation,
+        compression: Option<Compression>,
+    ) -> Result<Self, Error> {
+        let raw = if let Some(compression) = compression {
+            let compressed_len = u32::from_le_bytes(raw[..4].try_into().unwrap()) as usize;
+            let Some(compressed) = raw[4..].get(..compressed_len) else {
+                todo!()
```

Comment:
are the todos just to throw the "right" errors?

Reply:
Argh, in fact those are there because I forgot to implement them as proper errors. Will fix.

Reply:
I pushed a fix for this (and other errors found by CI).